packages/video_player/video_player/android/src/main/java/io/flutter/plugins/videoplayer/Messages.java

```diff
@@ -0,0 +1,338 @@
+      fromMapResult.isLooping = (bool)map.get("isLooping");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class VolumeMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    private Double volume;
+    public Double getVolume() { return volume; }
+    public void setVolume(Double setterArg) { this.volume = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      toMapResult.put("volume", volume);
+      return toMapResult;
+    }
+    static VolumeMessage fromMap(HashMap map) {
+      VolumeMessage fromMapResult = new VolumeMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      fromMapResult.volume = (double)map.get("volume");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class PositionMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    private Long position;
+    public Long getPosition() { return position; }
+    public void setPosition(Long setterArg) { this.position = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      toMapResult.put("position", position);
+      return toMapResult;
+    }
+    static PositionMessage fromMap(HashMap map) {
+      PositionMessage fromMapResult = new PositionMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      fromMapResult.position = (int)map.get("position");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated interface from Pigeon that represents a handler of messages from Flutter.*/
+  public interface VideoPlayerApi {
+    VoidMessage initialize(VoidMessage arg);
+    TextureMessage create(CreateMessage arg);
+    VoidMessage dispose(TextureMessage arg);
+    VoidMessage setLooping(LoopingMessage arg);
+    VoidMessage setVolume(VolumeMessage arg);
+    VoidMessage play(TextureMessage arg);
+    PositionMessage position(TextureMessage arg);
+    VoidMessage seekTo(PositionMessage arg);
+    VoidMessage pause(TextureMessage arg);
+
+    /** Sets up an instance of `VideoPlayerApi` to handle messages through the `binaryMessenger` */
+    public static void setup(BinaryMessenger binaryMessenger, VideoPlayerApi api) {
+      {
+        BasicMessageChannel<Object> channel =
+            new BasicMessageChannel<Object>(binaryMessenger, "dev.flutter.pigeon.VideoPlayerApi.initialize", new StandardMessageCodec());
+        channel.setMessageHandler(new BasicMessageChannel.MessageHandler<Object>() {
```

Comment:
I suspect we might have to do some state keeping at some point to let the use remove these handlers when time comes to clean up. I think you fixed some bugs recently on letting plugins unregister on iOS too.